js/source/source_cache.js

```diff
@@ -422,7 +422,7 @@ SourceCache.prototype = util.inherit(Evented, {
 
         tile.uses--;
         delete this._tiles[id];
-        this._source.fire('data', {dataType: 'tile'});
+        this._source.fire('data', { tile: tile, dataType: 'tile', tileRemoved: true });
```

Comment:
Would you be open to removing the `tileRemoved` property and branching on `tile.uses === 0` in your code to determine if the tile has been unloaded?